### ISSUE TITLE
.ref on mounted component returns HTML-Element instead of ReactWrapper

### ISSUE THREAD

**Issue by bdemolder** [airbnb/enzyme#1459]
### Current behavior

Imagine a component that renders a div with 3 anchor elements inside, each with their ref:
```
<div>
  <a ref="element1" onClick="..."></a>
  <a ref="element2" onClick="..."></a>
  <a ref="element3" onClick="..."></a>
</div>
```

And you want to simulate an onclick behavior in a jest test.

So I mount the component:
`var sut = mount(<Component>);`

And then when I get the element wit hthe '.ref()' function, it does not return a ReactWrapper, but an HTMLAnchorElement, which does not have a .simulate or .prop?

However, the following does return a nice ReactWrapper and works as I wanted, but is not nice:
`sut.find('a').at(0).simulate('click');`

My best guess is that I do something wrong, but please forgive me if it is something silly.

### Expected behavior

As the documentation says, return a ReactWrapper so I can act on it?

### Your environment

?

#### API

- [ ] shallow
- [X] mount
- [ ] render

#### Version

| library | version
| ---------------- | -------
| Enzyme | 3.3.0
| React | 16.2.0
| Jest |  16.2.0
| enzyme-adapter-react-16 | 1.1.1

#### Adapter

- [X] enzyme-adapter-react-16
- [ ] enzyme-adapter-react-15
- [ ] enzyme-adapter-react-15.4
- [ ] enzyme-adapter-react-14
- [ ] enzyme-adapter-react-13
- [ ] enzyme-adapter-react-helper
- [ ] others ( )

**Comment by ljharb** [comment]
Refs are DOM nodes, typically. They can also sometimes be React component instances, but they'd never be an enzyme wrapper in real code.

The [docs](http://airbnb.io/enzyme/docs/api/ReactWrapper/ref.html) definitely do say that `.ref()` returns a ReactWrapper; but string refs are deprecated - I'm not even sure if they work in react 16 at all. The proper solution is to use constructor-bound ref callbacks, and set a property on the instance.

**Comment by bdemolder** [comment]
True, I now vaguely remember that refs should not be string anymore, my bad, due all the upgrades of my packages I might have lost track of that part.

But thx! I will try that way, and otherwise I will just give them a proper ID tag. Sorry for the trouble.

**Comment by ljharb** [comment]
It is strange tho that they don’t match the docs; I’m going to reopen this to track that.

**Comment by xixixao** [comment]
It would be great to fix the docs:

https://github.com/airbnb/enzyme/blob/master/packages/enzyme/src/ReactWrapper.js#L184

And add this to the migration guide, as it is another breaking change between 2 and 3.

**Comment by Blasz** [comment]
I'd love to know how to fix this as manually wrapping a ref with `ReactWrapper` does not work as expected anymore in v3 as the ref's children are lost.
